### PR TITLE
Cli cmd unit tests

### PR DESCRIPTION
**Pull request by Wyvinar**

### CODE REVIEW

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-cli-pr-192-24xzm for details. If you are unable to do so, please contact a member of the Galasa team.

File: pkg/cmd/authLogin_test.go [hunk 1]
```diff
@@ -24,3 +26,62 @@ func TestAuthLoginCommandInCommandCollection(t *testing.T) {
...
+	// Then...
+
+	// Check what the user saw is reasonable.
+	stdOutConsole := factory.GetStdOutConsole().(*utils.MockConsole)
+	outText := stdOutConsole.ReadText()
+	assert.Contains(t, outText, "Displays the options for the 'auth login' command.")
+
+	stdErrConsole := factory.GetStdErrConsole().(*utils.MockConsole)
+	errText := stdErrConsole.ReadText()
+	assert.Empty(t, errText)
```
Comment (techcobweb) [inline pkg/cmd/authLogin_test.go:49]: These 3 sections are repeated many times. Perhaps a method like 
checkOutput(expectedStdOut, expectedStdErr, expectedFinalError) would compress this test code ?

File: pkg/cmd/propertiesGet_test.go [hunk 2]
```diff
@@ -53,4 +54,274 @@ func TestPropertiesGetHelpFlagSetCorrectly(t *testing.T) {
...
+	factory := NewMockFactory()
+	commandCollection, err := NewCommandCollection(factory)
+	assert.Nil(t, err)
+
+	var propertiesGetCommand GalasaCommand
+	propertiesGetCommand, err = commandCollection.GetCommand("properties get")
+	assert.Nil(t, err)
+	propertiesGetCommand.CobraCommand().RunE = func(cobraCmd *cobra.Command, args []string) error { return nil }
+
+	var args []string = []string{"properties", "get", "--namespace", "mince", "--name", "pies.are.so.tasty"}
```
Comment (techcobweb) [inline pkg/cmd/propertiesGet_test.go:86]: Totally agree. Making me hungry.

File: pkg/cmd/propertiesGet_test.go [hunk 3]
```diff
@@ -53,4 +54,274 @@ func TestPropertiesGetHelpFlagSetCorrectly(t *testing.T) {
...
+	assert.Nil(t, err)
+	propertiesGetCommand.CobraCommand().RunE = func(cobraCmd *cobra.Command, args []string) error { return nil }
+
+	var args []string = []string{"properties", "get", "--namespace", "mince"}
+
+	// When...
+	err = commandCollection.Execute(args)
+
+	// Then...
+	// We expect an exit code of 1 for this command. But it seems that syntax errors caught by cobra still return no error.
```
Comment (techcobweb) [inline pkg/cmd/propertiesGet_test.go:117]: The command is valid, so I wouldn't expect an error.

File: pkg/cmd/propertiesGet_test.go [hunk 4]
```diff
@@ -53,4 +54,274 @@ func TestPropertiesGetHelpFlagSetCorrectly(t *testing.T) {
...
+	assert.Nil(t, err)
+	propertiesGetCommand.CobraCommand().RunE = func(cobraCmd *cobra.Command, args []string) error { return nil }
+
+	var args []string = []string{"properties", "get", "--namespace", "mince", "--name", "pies.are.so.tasty", "--suffix", "something"}
+
+	// When...
+	err = commandCollection.Execute(args)
+
+	// Then...
+	// We expect an exit code of 1 for this command. But it seems that syntax errors caught by cobra still return no error.
```
Comment (techcobweb) [inline pkg/cmd/propertiesGet_test.go:168]: The syntax is valid so I wouldn't expect an error.

File: pkg/cmd/propertiesGet_test.go [hunk 5]
```diff
@@ -53,4 +54,274 @@ func TestPropertiesGetHelpFlagSetCorrectly(t *testing.T) {
...
+	assert.Nil(t, err)
+	propertiesGetCommand.CobraCommand().RunE = func(cobraCmd *cobra.Command, args []string) error { return nil }
+
+	var args []string = []string{"properties", "get", "--namespace", "mince", "--name", "pies.are.so.tasty", "--infix", "something"}
+
+	// When...
+	err = commandCollection.Execute(args)
+
+	// Then...
+	// We expect an exit code of 1 for this command. But it seems that syntax errors caught by cobra still return no error.
```
Comment (techcobweb) [inline pkg/cmd/propertiesGet_test.go:194]: According to cobra code, the command is valid.

File: pkg/cmd/propertiesGet_test.go [hunk 6]
```diff
@@ -53,4 +54,274 @@ func TestPropertiesGetHelpFlagSetCorrectly(t *testing.T) {
...
+	factory := NewMockFactory()
+	commandCollection, err := NewCommandCollection(factory)
+	assert.Nil(t, err)
+
+	var propertiesGetCommand GalasaCommand
+	propertiesGetCommand, err = commandCollection.GetCommand("properties get")
+	assert.Nil(t, err)
+	propertiesGetCommand.CobraCommand().RunE = func(cobraCmd *cobra.Command, args []string) error { return nil }
+
+	var args []string = []string{"properties", "get", "--namespace", "mince", "--suffix", "something"}
```
Comment (techcobweb) [inline pkg/cmd/propertiesGet_test.go:239]: what about when the --namespace, or --suffix parameters don't have a value ?

File: pkg/cmd/propertiesGet_test.go [hunk 7]
```diff
@@ -53,4 +54,274 @@ func TestPropertiesGetHelpFlagSetCorrectly(t *testing.T) {
...
+	factory := NewMockFactory()
+	commandCollection, err := NewCommandCollection(factory)
+	assert.Nil(t, err)
+
+	var propertiesGetCommand GalasaCommand
+	propertiesGetCommand, err = commandCollection.GetCommand("properties get")
+	assert.Nil(t, err)
+	propertiesGetCommand.CobraCommand().RunE = func(cobraCmd *cobra.Command, args []string) error { return nil }
+
+	var args []string = []string{"properties", "get", "--namespace", "mince", "--prefix", "something", "--suffix", "suffixthingy", "--infix", "infixthingy"}
```
Comment (techcobweb) [inline pkg/cmd/propertiesGet_test.go:289]: What if --namespace is used twice ?

Comment (techcobweb) [reply pkg/cmd/propertiesGet_test.go:289]: I'm wondering about another test.

**Review by techcobweb — APPROVED** [review]
A few minor comments, and the build is failing... but approving assuming you resolve those things.

File: pkg/cmd/authLogin_test.go (hunk 1 above)
Comment (techcobweb) [reply pkg/cmd/authLogin_test.go:49]: Or even testCommandCombination(args, expectedExecuteError, expectedStdOut, expectedStdErr, expectedFinalError)

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: pkg/cmd/authLogin_test.go (hunk 1 above)
Comment (Wyvinar) [reply pkg/cmd/authLogin_test.go]: made a function for the creation of the command collection and replacing of the the runE function as well.

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-cli-pr-192-9bxv8 for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build successful

File: pkg/cmd/propertiesSet_test.go [hunk 8]
```diff
@@ -24,3 +24,114 @@ func TestPropertiesSetCommandInCommandCollection(t *testing.T) {
...
+	// Then...
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "required flag(s) \"name\", \"namespace\", \"value\" not set")
+
+	// Check what the user saw is reasonable.
+	checkOutput("", "Error: required flag(s) \"name\", \"namespace\", \"value\" not set", "", factory, t)
+}
+
+//
+// DON"T KNOW WHY THIS TEST ISN'T WORKING
```
Comment (techcobweb) [inline pkg/cmd/propertiesSet_test.go:64]: Maybe we should discuss this one ?

**Review by techcobweb — APPROVED** [review]
Apart from the commented-out test, approved assuming that gets sorted out.

File: pkg/cmd/propertiesSet_test.go (hunk 8 above)
Comment (Wyvinar) [reply pkg/cmd/propertiesSet_test.go]: Discussed and fixed error

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**